src/app/scouter/page.tsx

```diff
@@ -0,0 +1,91 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import { ScouterViewer } from '@/components/ui/ScouterViewer';
+import { Button, createButtonProps } from '@/components/ui/Button';
+import { useState, useEffect } from 'react';
+import { getBattlePower } from '@/app/calc/ScouterCalculator';
+
+// 戦闘力計算用コンポーネント
+function ResultButton({ follower }: { follower: number }) {
+  const router = useRouter();
+  
+  const handleSubmit = () => {
+    const battlePower = getBattlePower(follower);
+    sessionStorage.setItem('battlePower', String(battlePower));
+    router.push('/result');
+  };
```

Comment:
ここも念の為失敗していないかはみておくとよさそうです

```suggestion
  const handleSubmit = () => {
    const battlePower = getBattlePower(follower);
    try {
      sessionStorage.setItem('battlePower',  String(battlePower));
      router.push('/result');
    } catch (e) {
      console.error("セッションストレージへの保存に失敗:", e);
      // ユーザーにエラーを通知する処理などをここに記述
    }
  };
```